keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1458 @@
+
+      * `pod.spec.resources.limits[memory/cpu] < pod.spec.containers[i].resources.limits[memory/cpu]` is invalid.
+
+        A single container’s limit should not be allowed to exceed the pod level limit.
+  4. Validation logic comparing requests and limits
+
+      1. Pod level requests greater than pod level limits is invalid.
+
+      2. Pod level requests greater than sum of limits of all containers is invalid.
+
+      3. Sum of container level requests greater than Pod level limits is invalid.
+        
+          This is already ensured by other validation rules in 1. and 2. as following:
+        
+          As per 1. Container level Requests <= Pod level requests is valid<br>
+          And as per 3.1 Pod-level requests <= Pod level limits is valid<br>
+          Hence, by transitivity these rules will ensure Container level requests <= Pod-level limits.
+
+      4. [Already exists] Sum of container level requests less than container level limits is invalid.
+
+#### Init Containers & Sidecar Containers
+
+With the current implementation (before this KEP), Pod’s effective resource request/limit is the sum of 
+pod overhead and the higher of:
+
+* The sum of all non-init containers(app and sidecar containers) request/limit for a resource.
+* The highest value resource request or limit among all the init containers.
+
+```
+Effective Request/Limit = Max(Max(Init container limit), Sum(Sidecar containers) + Sum (App Containers)) + Pod Overhead
+```
+
+With the support of pod-level requests and limits, following are a few options that can be used to determine pod’s effective resource request/limit:
+
+* **[Preferred] Option 1: Static values of effective pod request/limit as supplied in the PodSpec.**
+
+  In this the pod level cgroup will be set using the requests and limits from the resources specified at pod-level.
+
+  Example pod spec:
+
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  spec:
+    resources:
+      requests:
+        memory: 100Mi
+        cpu: 1
+      limits:
+        memory: 200Mi
+        cpu: 2
+    containers:
+      - name: container-1
+      - name: container-2
+        resources:
+          requests:
+            memory: 50Mi
+            cpu: 1
+  ```
+
+  **Resources requests at pod-level**
+
+  * The scheduler will use the requested resources (CPU and memory) settings specified in the PodSpec to find a suitable node to place the pod. 
+  * Kubelet will use the CPU requests from the pod-level resources section, and translate it to cpu.weight in pod-level cgroup.
+
+  **Resources limits at pod-level**
+
+  * All init(Regular and Sidecar) and non-init containers will be capped by the pod level limit which represents the peak usage of all containers.
+  * Kubelet will use CPU and memory limits from PodSpec, and translate them to cpu.max and memory.max cgroup settings at pod-level respectively. 
+
+  **Pros**
+
+  * Simple to understand and implement.
+  * Guarantees enough resources for all containers.
```

Comment:
this is a requirement in the current kubelet implementation. If we want to change this, I would suggest to run it as a separate KEP